counterparty-core/counterpartycore/lib/messages/utxo.py

```diff
@@ -160,9 +243,14 @@ def parse(db, tx, message):
 
     # detach if source is a UTXO
     if util.is_utxo_format(source):
-        source_address, _value = backend.bitcoind.get_utxo_address_and_value(source)
-        if source_address != tx["source"]:
-            problems.append("source does not match the UTXO source")
+        if util.enabled("spend_utxo_to_detach"):
+            utxos = tx["utxos_info"].split(" ")
+            if len(utxos) < 2 and source not in utxos[:-1]:
```

Comment:
shouldn't you throw the error if the utxo length is less than two OR the source is not in the input utxos? It looks like both conditions should invalidate the tx?

Am I correct in assuming the move_assets for attach/detach is moved until after parsing to ensure the above check clears before debiting/crediting the ledger?

Reply:
right! should be an OR. Fixed, thank you.

Yes right! see https://github.com/CounterpartyXCP/counterparty-core/pull/2626/files#diff-5af753d407909a443a7a31b3406b6eb0866c131536670744cf593ea0fd627ef0R120